src/main/scala/com/wavesplatform/UtxPool.scala

```diff
@@ -55,6 +89,23 @@ class UtxPool(
     }
   }
 
+  def portfolio(addr: Address): Portfolio = {
```

Comment:
With this approach we should go over all addresses in `UtxPool.removeExpired`:
```scala
      expiredTxs.foreach { tx =>
        transactions.remove(tx.id)
        txPessimisticPortfolios.remove(tx.id)
      }
```
but yes, I agree, that `removeExpired` could be executed rarely comparing with `portfolio` .